pennylane/qnode.py

```diff
@@ -151,7 +151,7 @@ class QNode:
             Only applies if the device is queried for the gradient; gradient transform
             functions available in ``qml.gradients`` are only supported on the backward
             pass. The 'best' option chooses automatically between the two options and is default.
-        mode (str): Whether the gradients should be computed on the forward
+        mode (str): Deprecated kwarg indicating whether the gradients should be computed on the forward
             pass (``forward``) or the backward pass (``backward``). Only applies
             if the device is queried for the gradient; gradient transform
             functions available in ``qml.gradients`` are only supported on the backward

```

Comment:
Should we mention that this should only be specified with `qml.disable_return()` called?

Reply:
`grad_on_execution` also works when `disable_return` is called.

Reply:
should I be doing that in this PR? I think I'd rather just leave it since it hasn't caused any trouble, and we're deleting it all next release anyway

Reply:
Yes I'm neutral towards the code changes, I just want to make sure that users know when reading this docstring that `mode` is pointless unless using the old return type.

Reply:
updated!